test/WebSites/CustomRouteWebSite/LocalizedRoute.cs

```diff
@@ -0,0 +1,64 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Microsoft.AspNet.Routing;
+using Microsoft.AspNet.Http;
+
+namespace CustomRouteWebSite
+{
+    public class LocalizedRoute : IRouter
+    {
+        private readonly IRouter _next;
+
+        private readonly Dictionary<string, string> _users = new Dictionary<string, string>(StringComparer.Ordinal)
+        {
+            { "Javier", "es-ES" },
+            { "Doug", "en-CA" },
+        };
+
+        public LocalizedRoute(IRouter next)
+        {
+            _next = next;
+        }
+
+        public string GetVirtualPath(VirtualPathContext context)
+        {
+            // We just want to act as a pass-through for link generation
+            return _next.GetVirtualPath(context);
+        }
+
+        public async Task RouteAsync(RouteContext context)
+        {
+            var oldRouteData = context.RouteData;
+
+            var newRouteData = new RouteData(oldRouteData);
+            newRouteData.Routers.Add(_next);
```

Comment:
This is the **stack of routers** feature. In System.Web you get the know *the route* that handled your request. However in vNext, there's a whole graph of `IRouter` that participates. So instead of a single answer you get all of the `IRouter` instances that are part of the current context.